client/components/public/imports/SamplePuzzles.jsx

```diff
@@ -40,33 +120,83 @@ export default class SamplePuzzles extends Component {
           <Grid padded centered textAlign="left" stackable>
             <Grid.Row>
               <Grid.Column width={16}>
-                Below You can find some of the past Puzzles. Download them and try to solve them for yourself.
+                Below you can find some of the past Puzzles. Download them and try to solve them for yourself!
               </Grid.Column>
-            </Grid.Row>
-            <Grid.Row centered columns={4}>
-                  { this._puzzles() }
-            </Grid.Row>
-
+            </Grid.Row>   
           </Grid>
+          { this._puzzles() }
           <Image fluid src="/img/2016/event-photos/team-the-purple-penguins-thin.jpg"/>
         </Segment>
        </Container>
-       
     )
   }
 
-
   _puzzles() {
-    return this.puzzles.map((puzzle, i) => (
-      <Grid.Column key={`puzzle-${i}`}>
-        <Segment  style={{minHeight:'250px'}} padded inverted color='blue' key={ puzzle.link }>
-          <Header as='h1' size="medium">{ puzzle.name }</Header>
-          { puzzle.tagline }
-          <br/>
-          <br/>
-          <Button  as='a' href='_blank' href={`/puzzles/2016/${puzzle.link}`} content='Download'/>
-        </Segment>
-      </Grid.Column>
+    const { activeIndex } = this.state;
+    let ridx = 0
+    return Object.keys(this.puzzles).map((year) => (
+      <Segment basic>
+      <Header as='h1'>{year}</Header>
+        <Button as='a' target='_blank' href={this.puzzles[year].infoPacket} content='Info Packet'/>
+        <br />
+        { this.puzzles[year].puzzles.map((puzzle, j) => (
+          <Accordion styled fluid>
+            <Accordion.Title active={activeIndex === ridx} index={ridx} onClick={(e,p) => this.handleClick(e,p)} >
+              <Icon color="black" size="huge" name="dropdown"/>
+              {j+1}. {puzzle.name}
+            </Accordion.Title>
+            <Accordion.Content active={activeIndex === ridx++}>
+              <Grid padded stackable centered textAlign="left">
+                <Grid.Row centered columns={4}>
+                  <Grid.Column>
+                    <Segment padded inverted color='blue'>
+                      <Header as='h2'>Puzzle</Header>
+                      <Button as='a' target='_blank' href={puzzle.link} content='Download'/>
+                    </Segment>
+                  </Grid.Column>
+                  <Grid.Column>
+                    <Segment padded inverted color='blue'>
+                      <Header as='h2'>Hints</Header>
+                      <List relaxed>
+                        { puzzle.hints.length ? puzzle.hints.map((file, k) => (
+                          <List.Item>
```

Comment:
Ah okay, actually THIS is what it's complaining about w/o a key. but the others may need keys too